src/_core/components/LayerMenu/LayerControlBase.js

```diff
@@ -0,0 +1,111 @@
+        this.isChangingPosition = false;
+        this.opacityButton = null;
+    }
+
+    shouldComponentUpdate(nextProps) {
+        // Here we prevent unnecessary renderings by explicitly
+        // ignoring certain pieces of the layer state. We do this
+        // since LayerControlBase is passed an entire layer object
+        // when instantiated in LayerMenuContainer, which contains state
+        // we want to ignore. By ignoring certain things, we can reduce
+        // the number of unnecessary renderings.
+        let nextLayer = nextProps.layer;
+        let currLayer = this.props.layer;
+        return (
+            nextProps.palette !== this.props.palette ||
+            nextLayer.get("title") !== currLayer.get("title") ||
+            nextLayer.get("opacity") !== currLayer.get("opacity") ||
+            nextLayer.get("isActive") !== currLayer.get("isActive") ||
+            nextLayer.get("palette") !== currLayer.get("palette") ||
+            nextLayer.get("min") !== currLayer.get("min") ||
+            nextLayer.get("max") !== currLayer.get("max") ||
+            nextLayer.get("units") !== currLayer.get("units") ||
+            nextLayer.get("displayIndex") !== currLayer.get("displayIndex")
+        );
+    }
+
+    setLayerActive(active) {
+        this.isChangingPosition = false;
+        this.isChangingOpacity = false;
+        this.props.mapActions.setLayerActive(this.props.layer.get("id"), !active);
+    }
+
+    changeOpacity(value) {
+        let opacity = value / 100.0;
+        this.props.mapActions.setLayerOpacity(this.props.layer, opacity);
+    }
+
+    toggleChangingOpacity() {
+        this.isChangingOpacity = !this.isChangingOpacity;
+        this.isChangingPosition = false;
+        this.forceUpdate();
+    }
+
+    toggleChangingPosition() {
+        this.isChangingPosition = !this.isChangingPosition;
+        this.isChangingOpacity = false;
+        this.forceUpdate();
+    }
+
+    openLayerInfo() {
+        this.props.mapActions.loadLayerMetadata(this.props.layer);
+    }
+
+    changePalette() {
+        this.props.mapActions.changeLayerPalette(this.props.layer.get("id"), {});
+    }
+
+    moveToTop() {
+        this.props.mapActions.moveLayerToTop(this.props.layer.get("id"));
+    }
+
+    moveToBottom() {
+        this.props.mapActions.moveLayerToBottom(this.props.layer.get("id"));
+    }
+
+    moveUp() {
+        this.props.mapActions.moveLayerUp(this.props.layer.get("id"));
+    }
+
+    moveDown() {
+        this.props.mapActions.moveLayerDown(this.props.layer.get("id"));
+    }
+
+    render() {
+        return <div className={displayStyles.hidden} />;
+    }
+}
+
+LayerControlBase.propTypes = {
+    mapActions: PropTypes.object.isRequired,
+    layer: PropTypes.object.isRequired,
+    activeNum: PropTypes.number.isRequired,
+    palette: PropTypes.object
+};
+
+LayerControlBase.mapDispatchToProps = function(dispatch) {
+    return {
+        mapActions: bindActionCreators(mapActions, dispatch)
```

Comment:
If we do split out the base component here do we want to pass in mapActions as a prop?

Reply:
I dunno. I like the idea of not having to deal with any part of state/dispatch mapping if you inherit from the base. But then we'd need a way to easily override actions downstream